Bottom: e75e5d4c9339f33641d5b8357bb241f944568525
Top:    d06309be5046fd59e64d2b94132a27ab595f47b4
Author: Ben Greear <greearb@candelatech.com>
Date:   2013-01-29 22:48:33 -0800

dev: Use sched-timeout instead of msleep(250)

Should help speed device removal in some cases.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/core/dev.c b/net/core/dev.c
index 51a105a..978ac8f 100644
--- a/net/core/dev.c
+++ b/net/core/dev.c
@@ -7259,7 +7259,7 @@ static void netdev_wait_allrefs(struct net_device *dev)
 			rebroadcast_time = jiffies;
 		}
 
-		msleep(250);
+		schedule_timeout_uninterruptible(1);
 
 		refcnt = netdev_refcnt_read(dev);
